op-batcher/batcher/channel_builder.go

```diff
@@ -175,22 +187,25 @@ func (c *channelBuilder) Reset() error {
 // AddBlock returns a ChannelFullError if called even though the channel is
 // already full. See description of FullErr for details.
 //
+// AddBlock also returns the L1BlockInfo that got extracted from the block's
+// first transaction for subsequent use by the caller.
+//
 // Call OutputFrames() afterwards to create frames.
-func (c *channelBuilder) AddBlock(block *types.Block) error {
+func (c *channelBuilder) AddBlock(block *types.Block) (derive.L1BlockInfo, error) {
 	if c.IsFull() {
-		return c.FullErr()
+		return derive.L1BlockInfo{}, c.FullErr()
 	}
 
-	batch, err := derive.BlockToBatch(block)
+	batch, l1info, err := derive.BlockToBatch(block)
 	if err != nil {
-		return fmt.Errorf("converting block to batch: %w", err)
+		return l1info, fmt.Errorf("converting block to batch: %w", err)
```

Comment:
Not necessarily in the scope of this pr, but defining a custom error type might help tracking errors in metrics and provide a cleaner way for comparison and externally interacting with the batcher.

Reply:
Custom Go error types should be used sparingly and only make sense if there’s a real chance that the caller receiving this error might react differently than to other errors. Otherwise they just clutter the package’s API. If you look e.g. at the standard library's packages, they only each provide a handful of custom errors, if at all, and only for those where there's a real chance the calling code might react to it. Good example is `io.EOF`, which is an "expected" error that the caller often reacts differently to. In this sense, error types are usually just used for "expected" errors that still represent a healthy program flow.

This example here is a good example of an error that does _not_ represent a healthy program flow. If the channel builder receives invalid block data, something is seriously wrong with the RPC client delivering blocks. I don't see a realistic scenario right now where the calling code would react to this differently. 

The default should be to just return `fmt.Errorf()` simple errors and only introduce a custom error type once _needed_. Otherwise it's premature optimization and clutters the package API.